### PR TITLE
enhance: optimize marisa trie range search for performance

### PR DESCRIPTION
**Pull request by zhagnlu**
#30078
#29986

### CODE REVIEW

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/core/src/common/Utils.h [hunk 1]
```diff
@@ -205,4 +205,21 @@ Join(const std::vector<T>& items, const std::string& delimiter) {
     return ss.str();
 }
 
+inline std::string
+GetCommonPrefix(const std::string& str1, const std::string& str2) {
+    std::string result;
```
Comment (czs007) [inline internal/core/src/common/Utils.h:210]: how about  reserve( len )  ?

Comment (zhagnlu) [reply internal/core/src/common/Utils.h:210]: not necessary, it is fast enough

Comment (czs007) [reply internal/core/src/common/Utils.h:210]: or you can first calcuate the len and return by str1.sbustr(0, len)

Comment (czs007) [reply internal/core/src/common/Utils.h:210]: size_t len = std::min(str1.length(), str2.length());
size_t i = 0;
while(i<len && str1[i] == str2[i]) ++i;
return str1.substr(0, i);

Comment (zhagnlu) [reply internal/core/src/common/Utils.h:210]: your implement is same if considering performance and resource use. it is  just more concise.

**Pull request merged**